### ISSUE TITLE
`TypeIs` should be treated as invariant

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#7615]
According to PEP 742, the `TypeIs` special form should be treated as invariant, not covariant.

The following code has a false negative:

```python
from typing import Callable
from typing_extensions import TypeIs


def takes_int_typeis(f: Callable[[object], TypeIs[int]]) -> None:
    pass


def int_typeis(val: object) -> TypeIs[int]:
    return isinstance(val, int)


def bool_typeis(val: object) -> TypeIs[bool]:
    return isinstance(val, bool)


takes_int_typeis(int_typeis)  # OK
takes_int_typeis(bool_typeis)  # Error
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.358, which I just published.
